java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -28,29 +45,51 @@ public String version() {
 
   @Override
   public void start(Map<String, String> props) {
-    log.info("Task initialising");
-    printAsciiHeader(jarManifest, "/azure-eventhubs-ascii.txt");
+    Optional<SourceTaskContext> contextOptional = ofNullable(context);
+    Map<String, String> configProps = null;
+    if (contextOptional.isPresent()) {
+      configProps = context.configs();
+      ofNullable(context.offsetStorageReader())
+          .ifPresent(TopicPartitionOffsetProvider::initialize);
+    }
 
-    Map<String, String> configProps;
-    boolean contextConfigEmpty = ofNullable(context).flatMap(
-        context -> ofNullable(context.configs()))
-      .flatMap(configs -> Optional.of(configs.isEmpty())).orElse(false);
-    if (contextConfigEmpty) {
+    if (configProps == null || configProps.isEmpty()) {
       configProps = props;
-    } else {
-      assert context != null;
-      configProps = context.configs();
     }
+
+    eventHubsKafkaConsumerController = new EventHubsKafkaConsumerController(
+        new AzureEventHubsConfig(configProps), new KafkaConsumerProvider(),
+        new ArrayBlockingQueue<>(BLOCKING_QUEUE_DEFAULT_CAPACITY));
```

Comment:
Is there a Jira to track this? The 20-items capacity is not a production-ready default. Both frameworks, depending on the message size can handle a lot more records per second. This option should also be configurable, or probably be a factor of max.poll.records settings. If the client can poll 1000 records in a poll, maybe the queue should be N x max.poll.records

Reply:
Let's keep in mind this is not 20 records: this is 20 PACKS of records (`ConsumerRecords`) class, not single `ConsumerRecord`). But I think it makes sense for it to be configurable, I added this.